frontend/src/employee-mobile-frontend/components/attendances/actions/MarkAbsentBeforehand.tsx

```diff
@@ -49,7 +56,12 @@ export default React.memo(function MarkAbsentBeforehand() {
   const [selectedAbsenceType, setSelectedAbsenceType] = useState<
     AbsenceType | undefined
   >(undefined)
-  const [, setAbsences] = useState<Result<Absence[]>>(Loading.of())
+  const [absences, setAbsences] = useState<Result<Absence[]>>(Loading.of())
+  const [absenceRanges, setAbsenceRanges] = useState<LocalDate[][]>([])
+  const [uiMode, setUiMode] = useState<
+    'default' | 'confirmDelete' | 'confirmExit'
+  >('default')
+  const [deleteRange, setDeleteRange] = useState<LocalDate[]>([])
```

Comment:
Doesn't seem like there is any reason to store this data in a `useState` hook, it should just be passed to the `deleteAbsences` function.

Reply:
We only delete the absences after we have clicked the confirm button in the modal. I added the deleteRange to the state so we know what absence range the user clicked in the list and what are we confirming in the modal. Can't figure out how to pass the information to the modal without storing it in the component state.

Reply:
Yes, sorry, you are absolutely right. I was thinking of a completely different way to do the confirmation modal, but it's not relevant here.